Mohamed gehad hussien

MGehad opened the pull request.

MGehad committed:
initial project

MGehad committed:
initial project

MGehad committed:
Add dependencies and SharedPref utility class

MGehad committed:
Refactor MainActivity layout and initialize SharedPref utility

MGehad committed:
Add navigation and view binding support in build configuration

MGehad committed:
Add login and registration functionality with UI components

MGehad committed:
Add splash screen and navigation setup for authentication flow

MGehad committed:
Implement user serialization and deserialization in SharedPref utility

MGehad committed:
Add authentication navigation and UI components

MGehad committed:
Add vector drawable resources for various UI components

MGehad committed:
Add Retrofit helper and service interface for meal API integration

MGehad committed:
Add data models for meal and meals response

MGehad committed:
Add recipe navigation and UI components, including fragments, layouts, and menu

Ahmed-Saber-25 commented on app/src/main/java/com/example/final_summer_course/features/views/recipe/RecipeItemDetailsFragment.kt:
```diff
@@ -0,0 +1,145 @@
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
```
kindly use navgraph and nav args for data transfering

Ahmed-Saber-25 commented on app/src/main/java/com/example/final_summer_course/features/views/recipe/RecipeItemDetailsFragment.kt:
```diff
@@ -0,0 +1,145 @@
+            meal.ingredient4 to meal.measure4,
+            meal.ingredient5 to meal.measure5,
+            meal.ingredient6 to meal.measure6,
+            meal.ingredient7 to meal.measure7,
```
use remote data fetching

MGehad committed:
DONE: "use navgraph and nav args for data transfering"

MGehad committed:
Add Room database implementation for meal management

MGehad committed:
Add favorite functionality and update UI components for meal details

MGehad committed:
Add FavoriteFragment and implement bottom navigation for recipe management

MGehad committed:
Add error handling for login and registration forms with validation messages

MGehad committed:
Refactor Meal data handling to use LiveData for real-time updates in FavoriteFragment and MealViewModel

MGehad committed:
Add background color to login and registration fragments; implement confirmation dialogs for logout and account deletion actions

MGehad committed:
Add FavoriteFragment with meal list display; update UI elements and navigation

MGehad committed:
Add SearchFragment for recipe search functionality; implement search input and results display

MGehad committed:
Update UI elements in fragments; adjust margins and add error handling for login

MGehad committed:
Implement search functionality in SearchFragment; update app name to "Recipe App"

MGehad committed:
Refactor favorite meal handling and streamline logout/account deletion processes

MGehad committed:
Refactor meal adapter initialization in fragments; implement submitList method for better data handling

MGehad committed:
Add UI screenshots for main app screens

Added screenshots for favorite, home, login, register, search, and splash screens to the screenshots directory for documentation and reference purposes.

MGehad committed:
Update README.md
